### PR TITLE
[JENKINS-57425] Run benchmarks only when a special Maven profile is defined

### PR DESCRIPTION
**Pull request by AbhyudayaSharma**
Run benchmarks separately from the main build
Follow-up to #63

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,2 +1,4 @@
 // Builds a module using https://github.com/jenkins-infra/pipeline-library
 buildPlugin(jenkinsVersions: [null, '2.150.2'])
+List<String> mvnOptions = ['test', '-Dbenchmark']
+infra.runMaven(mvnOptions)
```
Comment (oleg-nenashev) [inline Jenkinsfile]: Have you tested it locally? I do not think it is going to work. No node selection, for starters

Comment (AbhyudayaSharma) [reply Jenkinsfile]: No, I did not test it locally. I thought the pull request build would verify it.

**Comment by casz** [comment]
Please, these tests are incredibly slow. :scream:

**Comment by casz** [comment]
I would prefer if these tests were not run on pull requests unless requested, as a contributor, I would hate to sit and wait for my incremental build or waiting for tests to pass.

You could use a declarative pipeline to use a when block.

**Pull request merged**